### PR TITLE
LPS-158896 Look for the group id in the service context; fallback to thread local

### PR DESCRIPTION
**Pull request by adolfopa**

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/info/item/provider/FileEntryInfoItemFriendlyURLProvider.java [hunk 1]
```diff
@@ -79,6 +78,29 @@ public List<FriendlyURLEntryLocalization> getFriendlyURLEntryLocalizations(
...
+	private long _getGroupId() {
+		ServiceContext serviceContext =
+			ServiceContextThreadLocal.getServiceContext();
+
+		if (serviceContext == null) {
+			return GroupThreadLocal.getGroupId();
+		}
+
+		if (serviceContext.getThemeDisplay() == null) {
+			return serviceContext.getScopeGroupId();
```
Comment (AliciaGarciaGarcia) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/info/item/provider/FileEntryInfoItemFriendlyURLProvider.java:90]: This does not work if the site is **not** the same as the default.

https://user-images.githubusercontent.com/47524751/185404385-83129c98-d25e-4703-aebd-afc84ad8114e.mp4


This is more or less what we have before and did not work :(

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

**Comment by adolfopa** [comment]
@AliciaGarciaGarcia I had no other choice but to extend the friendly url servlet urls so that it accepts the site id explicitly. This is used now to detect duplicates and use the numeric fallback only when there's ambiguity (e.g. connected site and asset library, where both have a document with the same url title).

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Pull request closed without merging**